x/gamm/keeper/msg_server.go

```diff
@@ -46,6 +47,11 @@ func (server msgServer) CreateBalancerPool(goCtx context.Context, msg *balancer.
 	return &balancer.MsgCreateBalancerPoolResponse{PoolID: poolId}, err
 }
 
+func (server msgServer) MigrateSharesToFullRangeConcentratedPosition(goCtx context.Context, msg *balancer.MsgMigrateSharesToFullRangeConcentratedPosition) (*balancer.MsgMigrateSharesToFullRangeConcentratedPositionResponse, error) {
```

Comment:
Why do we need to methods for ` MigrateSharesToFullRangeConcentratedPosition` and `MigrateShares` here?

Looks like we could have just one

Reply:
Couldn't we say the same thing for CreateBalancerPool?

https://github.com/osmosis-labs/osmosis/blob/f0fccf345f46f4392c778accd153dacef96620e9/x/gamm/keeper/msg_server.go#L47-L50

I was just following previous design of extracting the primary logic in the msg Server, but I can change this if you think I should divert from previous design

Reply:
`CreateBalancerPool` exists because this what's defined by the message server interface:
https://github.com/osmosis-labs/osmosis/blob/df3853ff9886dad8604165f1fb43299d23caec2e/x/gamm/pool-models/balancer/tx.pb.go#L218-L220

Similar to `CreateStablesSwapPoll`. The reason we have `CreatePool` is to share logic between balancer and stableswap.

I'm not exactly sure why we would have redundant APIs with migrations

Reply:
Agreed, I misunderstood the purpose, changed!

https://github.com/osmosis-labs/osmosis/pull/4051/commits/b1008bc5da26c347b7969d4ecebe6c76c038a8ae